### DISCUSSION TITLE
v2.3.0-0

### DISCUSSION THREAD

**Discussion by eemeli** [eemeli/yaml#448] [General]
This release is made available first as a prerelease to gauge the effects of the [TypeScript changes](https://github.com/eemeli/yaml/pull/441), which add a second optional generic type argument `Strict` to `Document` instances. While this change appears to improve the TS usage experience at least with basic patterns, it's possible that it doesn't take into account some usage that would be negatively affected by the change.

Comments are requested, both for and against the change.

### New Features

- #441

### Bugfixes

- Use correct argument order when stringifying flow collection comments (#443)

<hr /><em>This discussion was created from the release <a href='https://github.com/eemeli/yaml/releases/tag/v2.3.0-0'>v2.3.0-0</a>.</em>

**Comment by swarnadeepsaha** [comment]
@eemeli 
unable to create Bug[Submit button disabled]
Hence mentioning here,

In this [change](https://github.com/eemeli/yaml/commit/78549261941e0efc4b98a5a7db63cd42acf0545e)
npm engine >=7 create an issue where npm >=6.14 <7.

**To Reproduce**
- node 14 minimum requirements (*npm is >=6.14*) with `engine-strict=true`.
https://nodejs.org/en/download/releases
- Or install node 14 using the *node version manager*(`nvm`) tool with `engine-strict=true`.  

`npm install` will fail with:
```
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for yaml@2.3.0: wanted: {"node":">= 14","npm":">= 7"} (current: {"node":"14.20.0","npm":"6.14.17"})
npm ERR! notsup Not compatible with your version of node/npm: yaml@2.3.0
npm ERR! notsup Not compatible with your version of node/npm: yaml@2.3.0
npm ERR! notsup Required: {"node":">= 14","npm":">= 7"}
npm ERR! notsup Actual:   {"npm":"6.14.17","node":"14.20.0"}
```

**Expected behavior**
npm >= 6.14, not npm >= 7

**Versions:**
- Environment: Node.js 14.20.0

**Reply by eemeli** [reply]
Please file this as an issue. The "submit" button is enabled after you add a title.

**Reply by swarnadeepsaha** [reply]
> Please file this as an issue. The "submit" button is enabled after you add a title.

https://github.com/eemeli/yaml/issues/476
